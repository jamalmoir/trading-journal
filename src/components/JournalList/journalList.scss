.journal-list {
  width: 100%;

  &> * {
    border: solid 1px #ced4da;
    border-top: none;
  }

  &> *:first-child {
    border-top: solid 1px #ced4da;
    border-top-left-radius: 0.25rem;
    border-top-right-radius: 0.25rem;
  }

  &> *:last-child {
    border-bottom-left-radius: 0.25rem;
    border-bottom-right-radius: 0.25rem;
  }

  &> *:nth-child(even) {
    background: #f8f9fa;
  }

  &> *:hover {
    background: #ebebeb;
  }
}