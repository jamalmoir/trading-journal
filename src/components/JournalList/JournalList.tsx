import * as React from 'react'
import Types from 'Types'
import './journalList.scss'
import { JournalCard } from '../JournalCard'

interface JournalListProps {
  className?: string
  journals: Types.Journal[]
}

const buildRows = (journals: Types.Journal[]) => {
  let rows = []

  for (let journal of journals) {
    rows.push(
      <JournalCard
        key={journal.id}
        journal={journal}
        className="cell small medium-4 large-4"
      />
    )
  }

  return rows
}

export const JournalList = (props: JournalListProps) => {
  return (
    <div
      className={
        'journal-list grid-x grid-margin-y grid-margin-x ' + props.className
      }
    >
      {buildRows(props.journals)}
    </div>
  )
}
