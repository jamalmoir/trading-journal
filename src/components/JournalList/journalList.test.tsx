
import * as React from 'react';
import { JournalList } from './JournalList';
import { shallow } from 'enzyme';
import Types from 'Types';


describe("<JournalList />", () => {
  it('should match snapshot', () => {
		const journals: Types.Journal[] = [
			{
				id: 'someId',
				userId: 'someUserId',
				kind: 'live',
				currency: 'GBP',
				name: 'Some Journal',
				created: new Date('2019/09/10'),
				modified: new Date('2019/09/10'),
				tradeCount: 10,
			},
			{
				id: 'someId2',
				userId: 'someUserId',
				kind: 'demo',
				currency: 'JPY',
				name: 'Some Journal2',
				created: new Date('2019/09/11'),
				modified: new Date('2019/09/11'),
				tradeCount: 22,
			},
		];
    const component = shallow(<JournalList journals={ journals }/>);
    expect(component).toMatchSnapshot();
  });
});