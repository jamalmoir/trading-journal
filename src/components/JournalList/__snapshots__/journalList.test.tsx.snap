// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<JournalList /> should match snapshot 1`] = `
<div
  className="journal-list undefined"
>
  <Component
    journal={
      Object {
        "created": 2019-09-09T15:00:00.000Z,
        "currency": "GBP",
        "id": "someId",
        "kind": "live",
        "modified": 2019-09-09T15:00:00.000Z,
        "name": "Some Journal",
        "tradeCount": 10,
        "userId": "someUserId",
      }
    }
    key="someId"
  />
  <Component
    journal={
      Object {
        "created": 2019-09-10T15:00:00.000Z,
        "currency": "JPY",
        "id": "someId2",
        "kind": "demo",
        "modified": 2019-09-10T15:00:00.000Z,
        "name": "Some Journal2",
        "tradeCount": 22,
        "userId": "someUserId",
      }
    }
    key="someId2"
  />
</div>
`;
